### PR TITLE
Feat/pixart lora

### PR DESCRIPTION
**Pull request by lawrence-cj**
# What does this PR do?
This PR 
1. Changes  the `scale_shift_table` in [transformer_2d.py](https://github.com/lawrence-cj/diffusers/blob/225775a334c8e22aeb05be5a3f46957fbf85d82d/src/diffusers/models/transformer_2d.py#L222C42-L222C42) and [attention.py](https://github.com/lawrence-cj/diffusers/blob/225775a334c8e22aeb05be5a3f46957fbf85d82d/src/diffusers/models/attention.py#L234C30-L234C30) from torch.nn.Parameter to torch.nn.Emebdding when `norm_type == "ada_norm_single"` , which make the `scale_shift_table` possible for peft.LoraConfig
2. Change the maximum token length for the T5 tokenizer in [Python_pixart_alpha.py](https://github.com/lawrence-cj/diffusers/blob/225775a334c8e22aeb05be5a3f46957fbf85d82d/src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py#L307C8-L307C8) to give the user the flexibility to change the maximum token length for testing or further training. The change of the T5 maximum token length does not affect the default performance. Even if the number of max tokens increases, the embedding of the first 120 tokens is the same as the default value. Therefore, the performance is not affected by default.

Cc: @sayakpaul , I'm wondering how to add something like `deprecated` to prevent the user's checkpoint from staying in the original version and reporting errors.

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thanks for the changes! But I am afraid these are a bit backwards-breaking. To ensure a seamless user experience, I propose the following:

* We add conditional support through CLI arguments to the weight conversion script so that users can still retrieve the earlier weights (i.e., weights without `scale_shift_table` being just an `nn.Parameter`). 
* We host the newly converted weights on the Hub with a newer name: "PixArt-alpha/PixArt-XL-2-1024-MS-v0.2". Open to other ideas for naming. 
* We also change the config of `norm_type`: "ada_norm_single_v0.2". So that we can appropriately condition the initialization within the blocks. 
* Rest of the changes seem okay to me. 

@patrickvonplaten WDYT?

**Comment by patrickvonplaten** [comment]
This is indeed a bit too breaking for now, I'm afraid. 2 comments:
- 1.) Do we know that applying LoRA to this scale-shift parameter is really that important?
- 2.) Is there a way we can apply LoRA to it without changing it's type?

**Comment by sayakpaul** [comment]
@patrickvonplaten for 2., see relevant discussions here: https://github.com/huggingface/peft/issues/1272.

**Comment by lawrence-cj** [comment]
Agree with your opinion. Rollback to the original pipeline with only T5 max token length changed.

File: src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py [hunk 1]
```diff
@@ -688,6 +689,7 @@ def __call__(
         callback_steps: int = 1,
         clean_caption: bool = True,
         use_resolution_binning: bool = True,
+        model_token_max_length: int = 120,
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:692]: should we maybe make this rather part of the pipeline config (`model_index.json`) ? Wdyt @sayakpaul

**Comment by lawrence-cj** [comment]
Change to https://github.com/huggingface/diffusers/pull/6789

**Pull request closed without merging**